pkg/resource/deploy/step.go

```diff
@@ -521,7 +532,10 @@ func (s *ReadStep) Apply(preview bool) (resource.Status, StepCompleteFunc, error
 	}
 
 	complete := func() { s.event.Done(&ReadResult{State: s.new}) }
-	return resource.StatusOK, complete, nil
+	if resourceError == nil {
+		return resourceStatus, complete, nil
+	}
+	return resourceStatus, complete, resourceError
```

Comment:
It's interesting that the semantics here are to fail a deployment if we ever `Read` a resource that's in an unhealthy state. After thinking about it I think this makes sense, but it's also a little strange since Pulumi wasn't involved in resources that get read via `ReadStep`s.

Reply:
We will need to try this out IRL and refine the experience. _Not_ doing an automatic refresh will probably not be tractable for Kubernetes because the computed values change so often.

Reply:
Just tested it, actually this doesn't seem to be true -- it does succeed. In the situation where a resource failed to initialize, and the user has fixed it, when they type `pulumi update`, they're warned about previous errors, but the update proceeds and succeeds.

Reply:
Ok consensus is that this is ok, after talking offline with @pgavlin, @swgillespie, and @lukehoban.